### ISSUE TITLE
issues with FindGo

### ISSUE THREAD

**Issue by sbinet** [FairRootGroup/FairRoot#383]
building with 7d66a66fb4a73e6a492027be967c4eb3c0d19d6c (via `AliBuild`) I get:

```
DEBUG:FairRoot:7d66a66f: CMake Warning at CMakeLists.txt:146 (find_package):
DEBUG:FairRoot:7d66a66f:   By not providing "Findgo.cmake" in CMAKE_MODULE_PATH this project has asked
DEBUG:FairRoot:7d66a66f:   CMake to find a package configuration file provided by "go", but CMake did
DEBUG:FairRoot:7d66a66f:   not find one.
DEBUG:FairRoot:7d66a66f: 
DEBUG:FairRoot:7d66a66f:   Could not find a package configuration file provided by "go" with any of
DEBUG:FairRoot:7d66a66f:   the following names:
DEBUG:FairRoot:7d66a66f: 
DEBUG:FairRoot:7d66a66f:     goConfig.cmake
DEBUG:FairRoot:7d66a66f:     go-config.cmake
```

applying the following diff:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b01e973..d02fe22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,7 @@ find_package(Protobuf)
 find_package(Msgpack)
 find_package(FlatBuffers)
 find_package(DDS)
-find_package(go)
+find_package(Go)
 
 If(WITH_DBASE)
   find_package(MYSQL)
```

got me:
```
DEBUG:FairRoot:7d66a66f: -- Looking for Go...
DEBUG:FairRoot:7d66a66f: -- Looking for Go... found GO 1.6.2
DEBUG:FairRoot:7d66a66f: -- Found Go: /usr/local/go/bin/go (found version "1.6.2") 
```

also, the way the `examples/advanced/GoTutorial/CMakeLists.txt` file is written, it will build the binaries `go-sink` and `go-processor` during `cmake configure` instead of during `make`.

my `cmake-fu` is still very weak.
how would one create a proper target for these binaries ?

**Comment by MohammadAlTurany** [comment]
Hi
The "Go" vs "go" issue could be due to my case-insensitive MAC which I used as I did this!! I will correct it!

 > the way the examples/advanced/GoTutorial/CMakeLists.txt file is written, it will build the binaries go-sink and go-processor during cmake configure instead of during make.

This was the easiest and fastest way I could add the stuff to CMAKE. It is not nice but I thought for the beginning it should be ok! 

> how would one create a proper target for these binaries ?

One could try to use add_custom_target for that, but I am not sure if this is the right way, we have to see! maybe you take a look to the discussion here:
http://stackoverflow.com/questions/7978517/how-do-i-get-cmake-to-work-with-the-go-programming-language
